### DISCUSSION TITLE
Support for user accounts

### DISCUSSION THREAD

**Discussion by GNULain** [Difegue/LANraragi#111] [Ideas]
Hey there, I was wondering if it would be possible to add support for user accounts other than the admin account? It would be nice to give people the ability to upload without them having access to the other options.

**Comment by Difegue** [comment]
It's certainly possible, but not in my to-do list for the time being.  

The current admin login system is dead simple, but you'd have to extend it in a bunch of places to implement user accounts:
* Add mechanism in options to add new accounts and set/modify their passwords
* Store password hashes in the database in the same way as the current admin pass
* Rework login page to accept username/password pairs instead of just a password currently
* Add extra routing for "non-admin" users that would allow access to e.g uploading 
* To be cleaner on that last point, add a way to set which permissions are given to each user (uploading, editing metadata, running plugins, etc)

**Comment by Difegue** [comment]
If an account system does end up being implemented, note that it'd also possible to then save user preferences server-side. (such as reader/index options and reading progress across the server)
